Source/Core/Core/IOS/ES/ES.cpp

```diff
@@ -242,6 +242,23 @@ bool ES::LaunchTitle(u64 title_id, bool skip_reload)
   // (supposedly when trying to re-open those files).
   DiscIO::NANDContentManager::Access().ClearCache();
 
+  u32 device_id;
+  constexpr u32 DEFAULT_DEVICE_ID = 0x0403ac68;
+  if (title_id == Titles::SHOP &&
+      (GetDeviceId(&device_id) != IPC_SUCCESS || device_id == DEFAULT_DEVICE_ID))
+  {
+    ERROR_LOG(IOS_ES, "Refusing to launch the shop channel with default device credentials");
+    CriticalAlertT("You cannot use the Wii Shop Channel without using your own device credentials."
```

Comment:
I'd prefer keeping it factually correct; otherwise this is going to cause confusion if the user somehow has a NAND dump but not the keys.bin file in the NAND root, and if/when we end up supporting NAND images without extraction (as credentials are in the OTP).